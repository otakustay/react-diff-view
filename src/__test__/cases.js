/* In case you are looking for type of hunk, see src/Hunk/index.js */
import {parseDiff} from '..';

export const basic = `diff --git a/src/__test__/index.test.jsx b/src/__test__/index.test.jsx
index 643c2f0..7883597 100644
--- a/src/__test__/index.test.jsx
+++ b/src/__test__/index.test.jsx
@@ -21,3 +21,3 @@ describe('basic test', () => {
     test('App renders correctly', () => {
-        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();
+        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();
     });`;

export const basicHunk = parseDiff(basic)[0].hunks[0];

export const multiple = `diff --git a/src/index1.js b/src/index1.js
index e69de29..d00491f 100644
--- a/src/index1.js
+++ b/src/index1.js
@@ -0,0 +1 @@
+1
diff --git a/src/index.js b/src/index2.js
index d00491f..0cfbf08 100644
--- a/src/index2.js
+++ b/src/index2.js
@@ -1 +1 @@
-1
+2
diff --git a/src/index3.js b/src/index3.js
index 0cfbf08..e69de29 100644
--- a/src/index3.js
+++ b/src/index3.js
@@ -1 +0,0 @@
-2`;

export const multipleHunk = parseDiff(basic)[0].hunks[0];
