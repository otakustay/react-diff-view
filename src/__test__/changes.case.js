
export const normalChange = {isNormal: true, lineNumber: 0, oldLineNumber: 0};
export const insertChange = {isInsert: true, lineNumber: 0, oldLineNumber: 0};
export const deleteChange = {isDelete: true, lineNumber: 0, oldLineNumber: 0};
