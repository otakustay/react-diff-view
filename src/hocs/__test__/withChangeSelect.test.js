import renderer from 'react-test-renderer';
import {withChangeSelect} from '..';
import {Diff, Hunk} from '../..';
import {basicHunk} from '../../__test__/cases';

const ComponentIn = () => <div />;

describe('withChangeSelect', () => {
    test('returns a hoc function', () => {
        expect(typeof withChangeSelect()).toBe('function');
    });

    test('hoc function return a component', () => {
        expect(typeof withChangeSelect()(ComponentIn)).toBe('function');
    });

    test('render component', () => {
        const ComponentOut = withChangeSelect()(ComponentIn);
        expect(renderer.create(<ComponentOut />).toJSON()).toMatchSnapshot();
    });
});

const DiffView = ({hunks, selectedChanges, onToggleChangeSelection}) => {
    const codeEvents = {
        onClick: onToggleChangeSelection
    };
    const renderHunk = hunk => (
        <Hunk
            key={hunk.content}
            hunk={hunk}
            codeEvents={codeEvents}
        />
    );

    // TODO modify document
    return (
        <Diff hunks={hunks} selectedChanges={selectedChanges} diffType="modify" viewType="split" >
            {hunks => hunks.map(renderHunk)}
        </Diff>
    );
};

describe('withChangeSelect usage in document', () => {
    test('EnhancedDiffView snapshot', () => {
        const EnhancedDiffView = withChangeSelect({multiple: false})(DiffView);
        expect(renderer.create(<EnhancedDiffView hunks={[basicHunk]} />).toJSON()).toMatchSnapshot();
    });

    test('EnhancedDiffView snapshot with multiple', () => {
        const EnhancedDiffView = withChangeSelect({multiple: true})(DiffView);
        expect(renderer.create(<EnhancedDiffView hunks={[basicHunk]} />).toJSON()).toMatchSnapshot();
    });

    // TODO use enzyme to test click
});
