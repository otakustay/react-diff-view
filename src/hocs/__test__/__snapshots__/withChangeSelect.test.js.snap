// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withChangeSelect render component 1`] = `<div />`;

exports[`withChangeSelect usage in document EnhancedDiffView snapshot 1`] = `
<table
  className="diff diff-split"
  onMouseDown={[Function]}
>
  <colgroup>
    <col
      className="diff-gutter-col"
    />
    <col />
    <col
      className="diff-gutter-col"
    />
    <col />
  </colgroup>
  <tbody
    className="diff-hunk"
  >
    <tr
      className="diff-line diff-line-normal"
    >
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        21
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            test('App renders correctly', () =&gt; {
      </td>
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        21
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            test('App renders correctly', () =&gt; {
      </td>
    </tr>
    <tr
      className="diff-line diff-line-compare"
    >
      <td
        className="diff-gutter diff-gutter-delete"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        22
      </td>
      <td
        className="diff-code diff-code-delete"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
                expect(renderer.create(&lt;App diffText={'deff'} /&gt;).toJSON()).toMatchSnapshot();
      </td>
      <td
        className="diff-gutter diff-gutter-insert"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        22
      </td>
      <td
        className="diff-code diff-code-insert"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
                expect(renderer.create(&lt;App diffText={'diff'} /&gt;).toJSON()).toMatchSnapshot();
      </td>
    </tr>
    <tr
      className="diff-line diff-line-normal"
    >
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        23
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            });
      </td>
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        23
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            });
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`withChangeSelect usage in document EnhancedDiffView snapshot with multiple 1`] = `
<table
  className="diff diff-split"
  onMouseDown={[Function]}
>
  <colgroup>
    <col
      className="diff-gutter-col"
    />
    <col />
    <col
      className="diff-gutter-col"
    />
    <col />
  </colgroup>
  <tbody
    className="diff-hunk"
  >
    <tr
      className="diff-line diff-line-normal"
    >
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        21
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            test('App renders correctly', () =&gt; {
      </td>
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        21
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            test('App renders correctly', () =&gt; {
      </td>
    </tr>
    <tr
      className="diff-line diff-line-compare"
    >
      <td
        className="diff-gutter diff-gutter-delete"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        22
      </td>
      <td
        className="diff-code diff-code-delete"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
                expect(renderer.create(&lt;App diffText={'deff'} /&gt;).toJSON()).toMatchSnapshot();
      </td>
      <td
        className="diff-gutter diff-gutter-insert"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        22
      </td>
      <td
        className="diff-code diff-code-insert"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
                expect(renderer.create(&lt;App diffText={'diff'} /&gt;).toJSON()).toMatchSnapshot();
      </td>
    </tr>
    <tr
      className="diff-line diff-line-normal"
    >
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        23
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            });
      </td>
      <td
        className="diff-gutter diff-gutter-normal"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        23
      </td>
      <td
        className="diff-code diff-code-normal"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
            });
      </td>
    </tr>
  </tbody>
</table>
`;
